### DISCUSSION TITLE
Can not work on plain SSH connection, is this a problem?

### DISCUSSION THREAD

**Discussion by golanguage** [helix-editor/helix#6419] [Q&A] [answered]
I create a normal SSH connection to Linux, and run debug version of Helix (master branch), got a crash:
`Error: The keyboard enhancement status could not be read within a normal duration`
My SSH console became a bit of abnormal also.

I think this is because of crossterm-rs querying keyboard enhancement support
```
fn read_supports_keyboard_enhancement_raw() -> Result<bool> {
    // This is the recommended method for testing support for the keyboard enhancement protocol.
    // We send a query for the flags supported by the terminal and then the primary device attributes
    // query. If we receive the primary device attributes response but not the keyboard enhancement
    // flags, none of the flags are supported.
    //
    // See <https://sw.kovidgoyal.net/kitty/keyboard-protocol/#detection-of-support-for-this-protocol>

    // ESC [ ? u        Query progressive keyboard enhancement flags (kitty protocol).
    // ESC [ c          Query primary device attributes.
    const QUERY: &[u8] = b"\x1B[?u\x1B[c";
```
 I'm not familiar with Helix, I just want to know, for Helix  this is a  feature or bug.

**Comment by the-mikedavis** [comment, marked as answer]
This is a bug: failure to get a response to the keyboard enhancement support query shouldn't be cause Helix to bail out of claiming the terminal. I posted https://github.com/helix-editor/helix/pull/6438 to fix this
